common/app/views/fragments/atoms/interactive.scala.html

```diff
@@ -0,0 +1,44 @@
+@(interactive: _root_.model.content.InteractiveAtom, shouldFence: Boolean)
+
+@body = {
+    <style>
+        @HtmlFormat.raw(interactive.css)
+    </style>
+    @HtmlFormat.raw(interactive.html)
+}
+
+@iframeBody = {
+    @body
+    <script>
+        function resize() {
+            window.frameElement.height = document.body.offsetHeight;
+        }
+
+        var fonts = [].slice.apply(window.parent.document.styleSheets)
+            .filter(function (sheet) { return sheet.ownerNode.className.indexOf("webfont") > - 1; })
+            .map(function (sheet) { return sheet.ownerNode.textContent; })
+            .join(' ');
+        var css = document.createElement('style');
+        css.textContent = fonts;
+        document.head.appendChild(css);
+
+        window.addEventListener('resize', resize);
+        resize();
+    </script>
+    @for(js <- interactive.mainJS) {
+        <script src="https://interactive.guim.co.uk/libs/require.js/2.3.2/require.min.js"></script>
+        <script>
+            require(['@js'], function (interactive) {
+                interactive.boot(document.body);
+            });
+        </script>
+    }
+}
+
+@if(shouldFence) {
+    <iframe class="interactive-atom" srcdoc="@iframeBody.toString"></iframe>
```

Comment:
The main reason I haven't done this yet is because I'm in two minds about the right solution. My first thought was just to edit fence to support resizing, but fence still has the 2-3 sec execution delay that most JS has on frontend.

Personally, I like that the iframes now resize instantly (less page janking), but I'm not sure if it goes against performance guidelines or something. If it doesn't then I'd probably just implement a simpler `srcdoc` fallback (without all the resizing that `fence` does).

I don't know very much about the implications here in terms of performance vs user experience, so would be great to chat in person.